### DISCUSSION TITLE
Codespaces-based PRs breaking maintainer merge commits for some reason

### DISCUSSION THREAD

**Discussion by sr229** [community/community#3902] [Codespaces] [answered]
We had a instance that Codespaces-based PRs mysteriously "break" merges that it creates only a branch of the same name into the upstream instead of merging the commits back to the fork, it seems to have happened as per https://github.com/ppy/osu-framework/pull/4438#discussion_r627746222 and another PR I authored on the same repo before this.

**Comment by 2percentsilk** [comment, marked as answer]
Hey @sr229,

How was this PR opened? Was it within the codespace or from the GitHub UI?
If it was from the GitHub UI, you may need to check the "Allow edits from maintainers" checkbox when you create the PR.

![image](https://user-images.githubusercontent.com/3174849/118718282-e99c7d80-b7db-11eb-815f-28c6cbcb5bf6.png)

**Reply by sr229** [reply]
This was opened via GitHub UI. Edits from maintainers was enabled.

**Reply by 2percentsilk** [reply]
Hi @sr229,

If the PR was opened through the GitHub.com UI (like my screenshot above), it's likely a PR issue or branch permissions issue. Apologies I can't be of more help her.
